osv2/bulk.go

```diff
@@ -141,20 +144,50 @@ func (r *BulkRequest) Do(ctx context.Context, client *opensearch.Client) (*opens
 		return nil, err
 	}
 
-	resp := BulkResponse{}
-
-	if err := json.Unmarshal(respBuf.Bytes(), &resp); err != nil {
+	resp, err := r.parseResponse(respBuf)
+	if err != nil {
 		return nil, err
 	}
 
 	return &opensearchtools.OpenSearchResponse[BulkResponse]{
 		StatusCode:        osResp.StatusCode,
 		Header:            osResp.Header,
-		Response:          resp,
+		Response:          *resp,
 		ValidationResults: vrs,
 	}, nil
 }
 
+// parseResponse parses the response based on the bulk response settings
+func (r *BulkRequest) parseResponse(respBuf bytes.Buffer) (*BulkResponse, error) {
```

Comment:
No need for this method to return a pointer.  We've intentionally structured the code in this package to return results by value so that things don't escape to the heap.

Also, since nothing uses `respBuf` as a `bytes.Buffer` this should accept `[]byte` instead.